pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1798,7 +1798,11 @@ func (l *LibvirtDomainManager) SignalShutdownVMI(vmi *v1.VirtualMachineInstance)
 	}
 
 	if domState == libvirt.DOMAIN_RUNNING || domState == libvirt.DOMAIN_PAUSED {
-		err = dom.ShutdownFlags(libvirt.DOMAIN_SHUTDOWN_ACPI_POWER_BTN)
+		flag := libvirt.DOMAIN_SHUTDOWN_DEFAULT
+		if vmi.Spec.Domain.Features != nil && vmi.Spec.Domain.Features.ACPI.Enabled != nil && *vmi.Spec.Domain.Features.ACPI.Enabled == true {
+			flag = libvirt.DOMAIN_SHUTDOWN_ACPI_POWER_BTN
+		}
+		err = dom.ShutdownFlags(flag)
```

Comment:
@iholder101   , shutdown  flag is set based on ACPI  is true or false

Reply:
so .. for x86/arm  flow is same as old.

Reply:
IIUC we shouldn't even get here if ACPI is not enabled for the VMI. Instead of calling `handleVMIShutdown()`, we would call `KillVirtualMachine()`.

Am I missing anything?

Reply:
https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainDestroyFlags

as per understanding , it sends SIGTERM first , if still exist then it endup calling SIGKILL . 
this might end up with data loss . 

if helperVmShutdown() call in  vm.go 
if tryGracefully.
      **handleVMIShutdown()**  - we are initiating gracefulShutdown process -  shudown send acpi/default signal siganl
    
 after Grace period expired  then we are calling 
 **KillVirtualMachine()**